packages/server-wallet/jest/custom-matchers.ts

```diff
@@ -59,27 +59,30 @@ expect.extend({
     received: ObjectiveResult[],
     expected: ObjectiveDoneResult['type']
   ) => {
-    const pass = received.length > 0 && received.every(async o => (await o.done).type === expected);
+    const doneResults = await Promise.all(received.map(r => r.done));
+
+    const pass = received.length > 0 && doneResults.every(d => d.type === expected);
     if (pass) {
-      const matchingObjectives = received.filter(async o => (await o.done).type === expected);
+      const matchingObjectives = await printObjectives(
+        received.filter(async o => (await o.done).type === expected)
+      );
       return {
         pass,
         message: () =>
-          `expected every objective to not be ${expected}. Failing objectives: ${printObjectives(
```

Comment:
Is this the correct message? Isn't this the condition for a valid pass?

Reply:
For the passing case, you return a `message` that would be used with `not`.
 IE : `expect(results).not.toHaveObjectiveDoneStatus('Success")` would return the `expected every objective to not be...` message.